scripts/i18next/generate.ts

```diff
@@ -0,0 +1,280 @@
+import { existsSync, mkdirSync, readdirSync, readFileSync, realpathSync, writeFileSync } from 'fs';
+import { dirname, join, relative } from 'path';
+import { execSync } from 'child_process';
+
+import prettier from 'prettier';
+
+const NODE_MODULES_DIRECTORY = join(process.cwd(), 'node_modules');
+const PLATFORMS_DIRECTORY = join(process.cwd(), 'platforms');
+
+type SimplePackageJSON = {
+  dependencies?: Record<string, string>;
+  devDependencies?: Record<string, string>;
+  peerDependencies?: Record<string, string>;
+};
+
+type Config = {
+  /**
+   * The path of the locales folder
+   */
+  path: string;
+  /**
+   * The translation output path with placeholders for locale and namespace
+   */
+  output: string;
+  /**
+   * Bundle the translations with the build output (resources)
+   */
+  bundle: boolean;
+  /**
+   * The path where the resources are bundled
+   */
+  bundlePath: string;
+  /**
+   * The path where the resources file is created
+   */
+  resourcesPath: string;
+};
+
+function getWorkspacePackages(packageJSON: SimplePackageJSON) {
```

Comment:
Does this function need to be recursive? I don't want to have to rely on the peer dependency model everywhere to surface 2nd+ level workspace dependencies.